src/runtime/runtime_nxpmk66f18.go

```diff
@@ -0,0 +1,400 @@
+	nxp.FTM0.C5SC.Set(0x28)
+	nxp.FTM0.C6SC.Set(0x28)
+	nxp.FTM0.C7SC.Set(0x28)
+
+	nxp.FTM3.C0SC.Set(0x28)
+	nxp.FTM3.C1SC.Set(0x28)
+	nxp.FTM3.C2SC.Set(0x28)
+	nxp.FTM3.C3SC.Set(0x28)
+	nxp.FTM3.C4SC.Set(0x28)
+	nxp.FTM3.C5SC.Set(0x28)
+	nxp.FTM3.C6SC.Set(0x28)
+	nxp.FTM3.C7SC.Set(0x28)
+
+	nxp.FTM0.SC.Set(nxp.FTM_SC_CLKS(1) | nxp.FTM_SC_PS(DEFAULT_FTM_PRESCALE))
+	nxp.FTM1.CNT.Set(0)
+	nxp.FTM1.MOD.Set(DEFAULT_FTM_MOD)
+	nxp.FTM1.C0SC.Set(0x28)
+	nxp.FTM1.C1SC.Set(0x28)
+	nxp.FTM1.SC.Set(nxp.FTM_SC_CLKS(1) | nxp.FTM_SC_PS(DEFAULT_FTM_PRESCALE))
+
+	// nxp.FTM2.CNT.Set(0)
+	// nxp.FTM2.MOD.Set(DEFAULT_FTM_MOD)
+	// nxp.FTM2.C0SC.Set(0x28)
+	// nxp.FTM2.C1SC.Set(0x28)
+	// nxp.FTM2.SC.Set(nxp.FTM_SC_CLKS(1) | nxp.FTM_SC_PS(DEFAULT_FTM_PRESCALE))
+
+	nxp.FTM3.CNT.Set(0)
+	nxp.FTM3.MOD.Set(DEFAULT_FTM_MOD)
+	nxp.FTM3.C0SC.Set(0x28)
+	nxp.FTM3.C1SC.Set(0x28)
+	nxp.FTM3.SC.Set(nxp.FTM_SC_CLKS(1) | nxp.FTM_SC_PS(DEFAULT_FTM_PRESCALE))
+
+	nxp.SIM.SCGC2.SetBits(nxp.SIM_SCGC2_TPM1)
+	nxp.SIM.SOPT2.SetBits(nxp.SIM_SOPT2_TPMSRC(2))
+	nxp.TPM1.CNT.Set(0)
+	nxp.TPM1.MOD.Set(32767)
+	nxp.TPM1.C0SC.Set(0x28)
+	nxp.TPM1.C1SC.Set(0x28)
+	nxp.TPM1.SC.Set(nxp.FTM_SC_CLKS(1) | nxp.FTM_SC_PS(0))
+
+	// configure the low-power timer
+	nxp.SIM.SCGC5.SetBits(nxp.SIM_SCGC5_LPTMR)
+	nxp.LPTMR0.CSR.Set(nxp.LPTMR0_CSR_TIE)
+	arm.EnableIRQ(nxp.IRQ_LPTMR0)
+
+	// 	analog_init();
+
+	// #if !defined(TEENSY_INIT_USB_DELAY_BEFORE)
+	// 	#if TEENSYDUINO >= 142
+	// 		#define TEENSY_INIT_USB_DELAY_BEFORE 25
+	// 	#else
+	// 		#define TEENSY_INIT_USB_DELAY_BEFORE 50
+	// 	#endif
+	// #endif
+
+	// #if !defined(TEENSY_INIT_USB_DELAY_AFTER)
+	// 	#if TEENSYDUINO >= 142
+	// 		#define TEENSY_INIT_USB_DELAY_AFTER 275
+	// 	#else
+	// 		#define TEENSY_INIT_USB_DELAY_AFTER 350
+	// 	#endif
+	// #endif
+
+	// 	// for background about this startup delay, please see these conversations
+	// 	// https://forum.pjrc.com/threads/36606-startup-time-(400ms)?p=113980&viewfull=1#post113980
+	// 	// https://forum.pjrc.com/threads/31290-Teensey-3-2-Teensey-Loader-1-24-Issues?p=87273&viewfull=1#post87273
+
+	// 	delay(TEENSY_INIT_USB_DELAY_BEFORE);
+	// 	usb_init();
+	// 	delay(TEENSY_INIT_USB_DELAY_AFTER);
+}
+
+func startupEarlyHook() {
+	// TODO allow override
+	// > programs using the watchdog timer or needing to initialize hardware as
+	// > early as possible can implement startup_early_hook()
+
+	nxp.WDOG.STCTRLH.Set(nxp.WDOG_STCTRLH_ALLOWUPDATE)
+}
+
+func startupLateHook() {
+	// TODO allow override
+}
+
+func putchar(c byte) {
+	u := &machine.UART1
+
+	// ensure the UART has been configured
+	if !u.SCGC.HasBits(u.SCGCMask) {
+		u.Configure(machine.UARTConfig{BaudRate: DefaultStdoutBaudRate})
+	}
+
+	for u.TCFIFO.Get() > 0 {
+		// busy wait
+	}
+	u.D.Set(c)
+}
+
+// ???
+const asyncScheduler = false
+
+// convert from ticks (us) to time.Duration (ns)
+const tickMicros = 1000
+
+var cyclesPerMilli = machine.CPUFrequency() / 1000
+var cyclesPerMicro = machine.CPUFrequency() / 1000000
```

Comment:
I wanted to define them once since they are used in multiple places. Once I fix the sleepWait calculations, `cyclesPerMicro` will only be used in one place. However `cyclesPerMilli` is used as the systick reset value and thus is needed when calculating ticks. I thought using the same variable would clarify the relationship between those two.